### ISSUE TITLE
Replace spaces for Unix output of wc in git_module

### ISSUE THREAD

**Issue by MitchLillie** [chris-marsh/pureline#17]
[This SO answer](https://stackoverflow.com/a/30927885/5213573) explains why OSX (and others) always output a set number of spaces from `wc`. As a result, git_module on OSX is almost unusable as-is, as the line contains many unnecessary spaces.

Proposal: 

Pipe the output of every `wc` command to `sed 's/^ *//'` in order to ensure spaces are trimmed.

**Comment by chris-marsh** [comment]
Can you give me some more information on this please?
The git module uses `wc -l `to count the lines returned. Leading spaces should not be an issue.

**Comment by i-need-to-tell-you-something** [comment]
@MitchLillie Is there any hacky workaround for this so I can use pureline on macOS?

**Comment by chris-marsh** [comment]
@i-need-to-tell-you-something , thanks. Your screenshot explains it well, I'll have a look into this.

**Comment by chris-marsh** [comment]
I'm thinking about replacing all use of 'wc -l' with this function ...
```# -----------------------------------------------------------------------------
# fix for macos where 'wc -l' returns the number within a padded string
# arg: input is piped in same fashion as wc / wc -l would be used
# example usage:
#   job_count=$(jobs -p | stripped_lc)
function stripped_lc {
    read piped_input
    local line_count=$(echo "$piped_input" | wc -l)
    echo "${line_count// /}"
}
```
Feedback if this works and I can go ahead and make the changes.

**Comment by i-need-to-tell-you-something** [comment]
I'm too beginner myself to know this. But if you tell me which files to edit or if I can pull your branch, then I'll gladly try it out. :)

**Comment by chris-marsh** [comment]
No problem. I will have a go this evening and push a branch for testing.

**Comment by MitchLillie** [comment]
Sorry for the delay in responding. Sounds like there is a good solution out there that @chris-marsh is working on.

@i-need-to-tell-you-something My *extremely* hacky solution is as follows, basically manually tapping into the various `wc` commands and piping to `sed`, as I initially suggested:

```
function custom_git_module {
        local git_branch=$(git rev-parse --abbrev-ref HEAD  2> /dev/null | sed -E 's/([a-zA-Z])?([a-zA-Z]+)?(\/?)(BPOR-[0-9]+)[a-zA-Z0-9-]+/\1\3\4/')
        if [ -n "$git_branch" ]; then
                local bg_color=$1
                local fg_color=$2
                local content="${symbols[git]} $git_branch$git"

                if [ -n "$3" -a -n "$(git status --porcelain)" ]; then
                        fg_color=$3
                fi

                local number_modified=$(git diff --name-only --diff-filter=M 2> /dev/null | wc -l | sed 's/^ *//' )
                if [ ! "$number_modified" -eq "0" ]; then
                        content+=" ${symbols[soft_separator]} ${symbols[plus]} $number_modified"
                fi

                local number_staged=$(git diff --staged --name-only --diff-filter=AM 2> /dev/null | wc -l | sed 's/^ *//')
                if [ ! "$number_staged" -eq "0" ]; then
                        content+=" ${symbols[soft_separator]} ${symbols[tick]} $number_staged"
                fi

                local number_conflicts=$(git diff --name-only --diff-filter=U 2> /dev/null | wc -l | sed 's/^ *//')
                if [ ! "$number_conflicts" -eq "0" ]; then
                        content+=" ${symbols[soft_separator]} ${symbols[cross]} $number_conflicts"
                fi

                local number_untracked=$(git ls-files --other --exclude-standard | wc -l | sed 's/^ *//')
                if [ ! "$number_untracked" -eq "0" ]; then
                        content+=" ${symbols[soft_separator]} ${symbols[untracked]} $number_untracked"
                fi

                PS1+=$(section_end $fg_color $bg_color)
                PS1+=$(section_content $fg_color $bg_color " $content ")
                __last_color=$bg_color
        fi
}
```

**Comment by i-need-to-tell-you-something** [comment]
@MitchLillie I tried your hack and it was nice and clean, but it didn't display stashes or commits (ahead/behind) any more. None of the special symbols and none of the numbers. Only the branch name. I tested this by checking out an earlier commit. Also I verified that my git stash had 5 stashes/commits.

![Screenshot 2019-08-27 at 11 28 21](https://user-images.githubusercontent.com/9165395/63754762-16995600-c8be-11e9-9543-537484c753b6.png)

@chris-marsh I applied the fix you last shared in this thread as well (having removed the custom_git_module from @MitchLillie ) and it didn't have too long module any more, but all of the git-related values for any repo and branch were displayed as "1". See attached image:

![Screenshot 2019-08-27 at 11 13 30](https://user-images.githubusercontent.com/9165395/63754394-76dbc800-c8bd-11e9-8257-6c266f992c45.png)

The combination of the two fixes yielded a result that looked exactly like the custom_git_module from the first image.

**Comment by andresrinivasan** [comment]
With a clean clone of the repo, I solved the issue by globally replacing

`wc -l`

with 

`wc -l | tr -d ' '`

**Comment by i-need-to-tell-you-something** [comment]
@andresrinivasan Seems to work for me as well:
![Screenshot 2019-11-20 at 11 06 48](https://user-images.githubusercontent.com/9165395/69225272-b9dccd00-0b86-11ea-8743-e9186113ab2a.png)
Now I just have to get used to all these "new" symbols. :smile:

**Comment by andresrinivasan** [comment]
Improved with bash globbing.

PR #41

**Comment by i-need-to-tell-you-something** [comment]
## Review from end-user perspective:
TL;DR: Everything in the PR seems to be working as intended.

#### On Ubuntu 18.04:
Using GNU bash version 4.4.20(1)-release (x86_64-pc-linux-gnu). 
Tested with my personal conf and with `pureline/configs/powerline_full_256col.conf`.
* No visual difference between PR and master.
#### On macOS 10.15.2:
Using GNU bash version 5.0.7(1)-release (x86_64-apple-darwin18.5.0) (which I might have installed via `homebrew`). 
Tested with my personal conf and with `pureline/configs/powerline_full_256col.conf`.
* The master branch (obviously still) has those extended empty spaces.
* The PR looks like master (and PR) look on Ubuntu.
* The PR looks exactly like the previous workaround.
